### ISSUE TITLE
Consistent return value question

### ISSUE THREAD

**Issue by b-dur** [trustpilot/node-eslint-config#20]
I have implemented our eslint config to our b2b core framework and in this process I discovered this "peculiar" rule called `consistent-return`

Example:
``` 
function (param) {
  if (param === false) {
    return;
  } else {
    // do something
    return param;
  }
}
```

To comply with this rule the empty `return` statement needs to return "something". So `return void 0;` is a valid expression. `return void 0;` basically returns undefined which `return;` also does.
I my opinion `return void 0;` looks more clutter than just plain `return;`.

Another example:

`var dateMinLimit = attr.dateMinLimit || void 0;`

Why assign `void 0` (undefined) if `attr.dateMinLimit` is undefined?

What do you guys think? Maybe the `return void 0;` is a symptom of a bad code style somewhere else in the code?

**Comment by b-dur** [comment]
@fernicolosi

**Comment by halfzebra** [comment]
@b-dur even though this rule has one very valid use case, I think it should be removed. Having to return `undefined` or using `void` operator clutters the code with weird looking logic.

Here's the case when this rule is actually useful:

```javascript
// eslint error
function doSomething(condition) {
    if (condition) {
        return true;
    }
}
```

**Comment by madsgodvinjensen** [comment]
I wouldn't mind removing it either. It would be nice to have a rule that makes it impossible to write functions that either return something or not based on flow inside the function, but that's not possible with this rule I guess?

**Comment by b-dur** [comment]
@madsgodvinjensen It actually does exactly that. It enforce you to either return a value or return no value.

The use case that @halfzebra is making would fail if the rule is set cause only one condition returns true the other (else) condition doesn't return anything.

**Comment by MartinMouritzen** [comment]
I personally think it's clutter. We don't want to end up having to write Java ;)

**Comment by TheLaz** [comment]
@b-dur : Forcing a return value even if it is `undefined` is an overkill, I don't see the problem ignoring that rule 👍

**Comment by b-dur** [comment]
Alright then lets vote. 
Add your reaction to this comment.
👍 for yes - let us remove that rule
👎 for no - we should keep the rule

**Comment by martinbuberl** [comment]
@b-dur Not feeling strong about that rule - I'm okay with removing it 🔥 .

Here's another take on that rule and why it might be useful (but again, not feeling it):
https://stackoverflow.com/a/36209076

**Comment by b-dur** [comment]
@martinbuberl Exactly. When that linter error happens it could mean a bad coding practice was applied and one should rethink on how to structure the code slightly different.

**Comment by martinbuberl** [comment]
@b-dur Just following up on this issue. I believe the votes have spoken. Shall we remove that rule, or can we close the issue?

**Comment by b-dur** [comment]
@martinbuberl Let's remove the rule and close this issue

**Comment by b-dur** [comment]
@martinbuberl Could you CR #22

**Comment by b-dur** [comment]
Resolved in PR #22
